crates/ika-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -529,7 +529,9 @@ impl DWalletMPCManager {
                     let mut session_clone = session.clone();
                     session_clone
                         .serialized_full_messages
-                        .truncate(session.pending_quorum_for_highest_round_number);
+                        .retain(|round_number, _| {
+                            round_number < &session.pending_quorum_for_highest_round_number
+                        });
```

Comment:
Ok i'm not sure about this one: we should never accept messages for a round before we had quorum, so there shouldn't be any more messages here. However, if we get a threshold not reached, it could be that some one (maliciously) submitted messages for the next round. In that case, I guess we should remove those & report them as malicious. So please add that. And then we can remove this `retain` I believe